hmda/src/main/scala/hmda/validation/rules/lar/validity/common/V628_1.scala

```diff
@@ -22,7 +22,10 @@ object V628_1 extends EditCheck[LoanApplicationRegister] {
                               EthnicityNotApplicable)
 
   override def apply(lar: LoanApplicationRegister): ValidationResult =
-    when(lar.applicant.ethnicity.otherHispanicOrLatino is empty) {
-      lar.applicant.ethnicity.ethnicity1 is containedIn(validEthnicities)
+    if(lar.applicant.ethnicity.otherHispanicOrLatino.isEmpty) {
+      lar.applicant.ethnicity.ethnicity1 is containedIn(validEthnicities) and (lar.applicant.ethnicity.ethnicity1 not equalTo(EmptyEthnicityValue))
```

Comment:
You can remove `and (lar.applicant.ethnicity.ethnicity1 not equalTo(EmptyEthnicityValue)` since it's redundant